src/main/java/org/springframework/integration/aws/inbound/kinesis/KinesisMessageDrivenChannelAdapter.java

```diff
@@ -784,6 +788,37 @@ public void run() {
 			};
 		}
 
+		private GetRecordsResult getRecords(GetRecordsRequest getRecordsRequest) {
+			try {
+				GetRecordsResult result = KinesisMessageDrivenChannelAdapter.this.amazonKinesis
+					.getRecords(getRecordsRequest);
+				return result;
+			}
+			catch (ExpiredIteratorException e) {
+				if (logger.isDebugEnabled()) {
+					logger.debug("Shard iterator " + getRecordsRequest.getShardIterator() + " expired");
```

Comment:
The `info` would be better. Just in case some one would be interested in the log that inform about this situation.
Would be great to add a message in that log that a new shard iterator will be obtained from the stored checkpoint.
WDYT?